app/models/sources/digital_ocean.rb

```diff
@@ -1,14 +1,123 @@
 module Sources
-  class DigitalOcean < Source
+  class DigitalOcean < Sources::Fog
     class << self
       def required_fields
-        {
-          username: :string,
-          password: :password,
-          image_name: :string
-        }
+        super.merge({
+          client_id: :string,
+          api_key: :password,
+          image_name: :string,
+          flavor_name: :string,
+          region_name: :string
+        })
       end
     end
+
+    ID_TYPES = ['image', 'flavor', 'region']
+
+    def valid_config?
```

Comment:
This is a potentially destructive operation, because `#names_to_ids` calls `#save`. Perhaps rename this `validate_config!`?